### PR TITLE
#585 Add parameter for the coeffs type in TaylorModelReachSet

### PR DESCRIPTION
**Pull request by dfcaporale**
Expands TaylorModelReachsets such that the coefficients can be intervals.

### CODE REVIEW

File: src/ReachSets/TaylorModelReachSet.jl [hunk 1]
```diff
@@ -55,7 +55,8 @@ function project(R::TaylorModelReachSet, vars::NTuple{D, M}) where {D, M<:Intege
 end
 
 # constructor with a time point
-function TaylorModelReachSet(X::Vector{TaylorModel1{TaylorN{N}, N}}, t::Real) where {N}
+#function TaylorModelReachSet(X::Vector{TaylorModel1{TaylorN{N}, N}}, t::Real) where {N}
```
Comment (mforets) [inline src/ReachSets/TaylorModelReachSet.jl:58]:
```suggestion
```

File: test/models/embrake.jl [hunk 2]
```diff
@@ -1,116 +0,0 @@
-set_rtol(Float64, 1e-12)
```
Comment (mforets) [inline test/models/embrake.jl:1]: removing this file seems unrelated

File: test/reachsets.jl [hunk 3]
```diff
@@ -102,4 +102,27 @@ end
     @test isequivalent(set(overapproximate(c, Zonotope)), set(R))
     @test isequivalent(set(overapproximate(d, Zonotope)), set(R))
 
+    order = 4
```
Comment (mforets) [inline test/reachsets.jl:105]: perhaps wrap this into its own test set, `@testset "Taylor model reach-sets with non-float coefficients" begin

Comment (mforets) [reply test/reachsets.jl:105]: see [CI log](https://github.com/JuliaReach/ReachabilityAnalysis.jl/runs/6041561496?check_suite_focus=true#step:6:618) for other errors; in particular, we need to make all the necessary `using`'s

Comment (dfcaporale) [reply test/reachsets.jl:111]: Thanks!

File: test/models/embrake.jl (hunk 2 above)
Comment (mforets) [reply test/models/embrake.jl:1]: better to resolve after it's addressed so we don't forget

File: test/reachsets.jl [hunk 4]
```diff
@@ -101,5 +101,36 @@ end
     @test tspan(c) == tspan(d) == 0..1
     @test isequivalent(set(overapproximate(c, Zonotope)), set(R))
     @test isequivalent(set(overapproximate(d, Zonotope)), set(R))
+                    
+end
 
+@testset "Taylor model reach-sets with non-float coefficients" begin
+    
+    using TaylorModels:Taylor1
```
Comment (mforets) [inline test/reachsets.jl:109]: by convention, the imports are added at the header of the file (see other files for examples)

File: test/reachsets.jl [hunk 5]
```diff
@@ -101,5 +101,36 @@ end
...
+    t = Taylor1(order+1)
+    
+    A = [0 0..0.1; 0..0.2 0..0.3]*t
+    n = size(A, 1)
+    R = ReachSet(X0, 0 .. 1)
+    RTM = overapproximate(R, TaylorModelReachSet)
+    X0_tm = set(RTM)
+    C = Vector{Vector{TaylorN{IntervalArithmetic.Interval{Float64}}}}(undef, n)
+    Y = [X.pol.coeffs[1] for X in X0_tm]
+    for i in 1:n
```
Comment (mforets) [inline test/reachsets.jl:122]: since `n = 2` here i think it'd be more clear if we just write down the defining equations of `C` instead of loops -- if you prefer the general form, then i suggest to write a helper function

File: test/reachsets.jl (hunk 4 above)
Comment (mforets) [inline test/reachsets.jl:109]: this is not the only missing name (see [here](https://github.com/JuliaReach/ReachabilityAnalysis.jl/runs/6047778975?check_suite_focus=true#step:6:620), note it only shows the next missing one) maybe try to run the tests locally first

File: test/models/embrake.jl (hunk 2 above)
Comment (mforets) [reply test/models/embrake.jl:1]: oh, i see now -- you want to delete this file because it's a duplicate with `EMBrake.jl`  and we only load the latter in the tests?

File: test/reachsets.jl [hunk 6]
```diff
@@ -103,3 +105,28 @@ end
     @test isequivalent(set(overapproximate(d, Zonotope)), set(R))
 
 end
+
+@testset "Taylor model reach-sets with non-float coefficients" begin
```
Comment (mforets) [inline test/reachsets.jl:109]: maybe a matter of taste, but to me it's cleaner if we create the object we want to test directly:

```julia
Δt, orderT, orderQ = 0..1, 4, 3
set_variables(IA.Interval{Float64}, "x", order=orderQ, numvars=2)
p1 = Taylor1([0, (0..0.1) + (0..0.01)*x[2]], orderT)
p2 = Taylor1([0, (0..0.5) + (0..0.02)*x[1] + (0..0.03)*x[2]], orderT)
vec = [TaylorModel1(p1, zeroI, zeroI, Δt), TaylorModel1(p2, zeroI, zeroI, Δt)]
T = TaylorModelReachSet(vec, Δt)
H = set(overapproximate(T, Hyperrectangle))

@test isa(T, TaylorModelReachSet) && isa(H, Hyperrectangle)
```

Comment (dfcaporale) [reply test/reachsets.jl:109]: You are right.It looks more clear this way!

File: test/models/embrake.jl (hunk 2 above)
Comment (dfcaporale) [reply test/models/embrake.jl:1]: I had my doubts with that...

**Review by mforets — APPROVED** [review]

**Comment by mforets** [comment]
Thanks!

This change may affect downstream packages, so I'll tag a new release.

**Comment by dfcaporale** [comment]
Thanks for your comments and reviews!
